### PR TITLE
Allow php8

### PR DESCRIPTION
**Pull request by snapshotpl**

### CODE REVIEW

**Comment by Nexucis** [comment]
I'm not sure it's ok to merge it since the lib [elasticsearch/elasticsearch](https://github.com/elastic/elasticsearch-php/blob/master/composer.json) doesn't support it currently.

**Comment by snapshotpl** [comment]
You have right

**Comment by snapshotpl** [comment]
Looks like PR for es lib for PHP8 already exists https://github.com/elastic/elasticsearch-php/pull/1063

**Comment by Nexucis** [comment]
yeah it doesn't look it's a top priority on elasticsearch side :/. Maybe if more people are complaining about this issue, it will go faster.

In the meantime, I was looking to the CI status to see what it is failing. It looks like the code / the tests are a bit outdated. Do you want to refresh it ? (Would suggest to do it in another PR)

**Comment by snapshotpl** [comment]
I'm not familiar with Circle CI so I can help with test too much. Looks like now CI run only on PHP 7.2, right?

**Comment by Nexucis** [comment]
yes that's right. 

But that's something you can try on your own environment I believe. After looking at the different issue, it seems it's because of the new version of PHP and also because of the new version of PHPUnit that causes the trouble.

The setup of circleci is not really important at this point.  I believe this is something quite easy to reproduce on a local environment :)

**Comment by Nexucis** [comment]
ah looks like elasticsearch is supporting now php8.

**Comment by Nexucis** [comment]
I will wait that elasticsearch is releasing a new version that supports php 8. And after I think we will be good

**Comment by snapshotpl** [comment]
@Nexucis elasticsearch is out https://github.com/elastic/elasticsearch-php/releases/tag/v7.11.0

**Comment by Nexucis** [comment]
\o/ nice. Do you want to finish this PR ?
On my side I won't be able to work on it before a couple of weeks I think.

**Comment by snapshotpl** [comment]
@Nexucis finally works! :-)

File: src/Nexucis/Elasticsearch/Helper/Nodowntime/IndexHelper.php [hunk 1]
```diff
@@ -240,7 +240,7 @@ public function addSettings($alias, $settings)
      * This method must call when you want to delete something inside the settings.
      *
      * @param string $alias [REQUIRED]
-     * @param array $settings [REQUIRED]
+     * @param mixed $settings [REQUIRED]
```
Comment (Nexucis) [inline src/Nexucis/Elasticsearch/Helper/Nodowntime/IndexHelper.php]: `mixed[]` no ?

File: src/Nexucis/Elasticsearch/Helper/Nodowntime/IndexHelper.php [hunk 2]
```diff
@@ -520,11 +520,11 @@ public function updateDocument($index, $id, $type, $body, $refresh = false)
 
     /**
      * @param string $index [REQUIRED] If the alias is associated to an unique index, you can pass an alias rather than an index
-     * @param $id [REQUIRED]
+     * @param string $id [REQUIRED]
      * @param string $type [REQUIRED]
-     * @param array $body [REQUIRED] : actual document to create
+     * @param mixed $body [REQUIRED] : actual document to create
```
Comment (Nexucis) [inline src/Nexucis/Elasticsearch/Helper/Nodowntime/IndexHelper.php]: `mixed[]` too ?

File: src/Nexucis/Elasticsearch/Helper/Nodowntime/IndexHelper.php [hunk 3]
```diff
@@ -700,20 +702,14 @@ protected function existsAlias($alias)
             'name' => urlencode($alias)
         );
 
-        $response = $this->client->indices()->existsAlias($params);
-
-        if (is_array($response) && array_key_exists('status', $response)) {
```
Comment (Nexucis) [inline src/Nexucis/Elasticsearch/Helper/Nodowntime/IndexHelper.php:705]: a bit weird you removed this code and it is still working. Was it a dead code ? Or am I missing something new in ElasticSearch ?

File: tests/Nexucis/Tests/Elasticsearch/Helper/Nodowntime/AbstractIndexHelperTest.php [hunk 4]
```diff
@@ -24,17 +24,19 @@ abstract class AbstractIndexHelperTest extends TestCase
     /**
      * initialize static data
      */
-    public static function setUpBeforeClass()
+    public static function setUpBeforeClass(): void
     {
-        // load static data
-        self::$documents = json_decode(file_get_contents('http://data.consumerfinance.gov/api/views.json'));
+        self::$documents = self::$documents === null
```
Comment (Nexucis) [inline tests/Nexucis/Tests/Elasticsearch/Helper/Nodowntime/AbstractIndexHelperTest.php:29]: definitely better like that indeed. Thanks for changed that !

File: tests/Nexucis/Tests/Elasticsearch/Helper/Nodowntime/IndexActionTest.php [hunk 5]
```diff
@@ -2,8 +2,13 @@
 
 namespace Nexucis\Tests\Elasticsearch\Helper\Nodowntime;
 
+use Nexucis\Elasticsearch\Helper\Nodowntime\Exceptions\IndexAlreadyExistException;
+use Nexucis\Elasticsearch\Helper\Nodowntime\Exceptions\IndexNotFoundException;
+use Yoast\PHPUnitPolyfills\Polyfills\AssertionRenames;
+
 class IndexActionTest extends AbstractIndexHelperTest
 {
+    use AssertionRenames;
```
Comment (Nexucis) [inline tests/Nexucis/Tests/Elasticsearch/Helper/Nodowntime/IndexActionTest.php:11]: just for my knowledge, what does it do or why is it needed ?

**Review by Nexucis — COMMENTED** [review]
Thanks a lot for the changes @snapshotpl !
I just added some comments, there are more questions that actually required changes.

File: tests/Nexucis/Tests/Elasticsearch/Helper/Nodowntime/IndexActionTest.php (hunk 5 above)
Comment (snapshotpl) [reply tests/Nexucis/Tests/Elasticsearch/Helper/Nodowntime/IndexActionTest.php:11]: To support old phpunit for php 7.1 and 7.2

File: src/Nexucis/Elasticsearch/Helper/Nodowntime/IndexHelper.php (hunk 3 above)
Comment (snapshotpl) [reply src/Nexucis/Elasticsearch/Helper/Nodowntime/IndexHelper.php:705]: `existsAlias` always return bool, so it was a dead code

File: tests/Nexucis/Tests/Elasticsearch/Helper/Nodowntime/AbstractIndexHelperTest.php (hunk 4 above)
Comment (snapshotpl) [reply tests/Nexucis/Tests/Elasticsearch/Helper/Nodowntime/AbstractIndexHelperTest.php:29]: Faster and stable! :-)

**Review by Nexucis — APPROVED** [review]
all good, thanks @snapshotpl !

**Pull request merged**